native/database/db.go

```diff
@@ -0,0 +1,124 @@
+package database
+
+import (
+	"database/sql"
+	"fmt"
+
+	_ "github.com/mattn/go-sqlite3" // Import the SQLite driver
+)
+
+// TransactionStatus represents the different states of a transaction.
+type TransactionStatus string
+
+// different states
+const (
+	StatusPending     TransactionStatus = "pending"
+	StatusBroadcasted TransactionStatus = "broadcasted"
+	StatusConfirmed   TransactionStatus = "confirmed"
+)
+
+// Transaction represents a transaction record in the database.
+type Transaction struct {
+	BtcTxID uint64            `json:"txid"`
+	RawTx   string            `json:"rawtx"`
+	Status  TransactionStatus `json:"status"`
+	// TODO: other fields
+}
+
+var db *sql.DB
+
+// InitDB initializes the database connection and creates the table if it doesn't exist
+func InitDB(dbPath string) error {
+	var err error
+	db, err = sql.Open("sqlite3", dbPath)
+	if err != nil {
+		return fmt.Errorf("failed to open database: %w", err)
+	}
+
+	_, err = db.Exec(`
+        CREATE TABLE IF NOT EXISTS transactions (
+            txid TEXT PRIMARY KEY,
+            rawtx TEXT NOT NULL,
+            status TEXT NOT NULL
+        )
+    `)
+	if err != nil {
```

Comment:
Why we don't return error directly here?

Reply:
Maybe I should use the logger there to log the error and just return it? What do you think?

Reply:
I think return error more better in this case.

Reply:
I wrapped the error to provide more details. But if you think just returning the error is better I will fix it

Reply:
Done